core/generated-sources/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/replace.kt

```diff
@@ -37,7 +37,7 @@ public fun <T, C> DataFrame<T>.replace(columns: Iterable<ColumnReference<C>>): R
 
 public fun <T> DataFrame<T>.replaceAll(
     vararg valuePairs: Pair<Any?, Any?>,
-    columns: ColumnsSelector<T, *> = { allDfs() },
+    columns: ColumnsSelector<T, *> = { all().recursively(includeGroups = false) },
```

Comment:
Why has the default value for includeGroups changed?

Reply:
Is it correct that it's because of unification of overloads?

Reply:
Yes, by default groups were included in dfs {}, but excluded in allDfs() if I recall correctly. I had to choose a sensible default. But if you use colsOf<Int>().rec(), then includeGroups irrelevant anyways, since Int is not a Group.

Reply:
Hmm, this is interesting and a bit confusing. I think this parameter only exists for allDfs. For normal dfs it would be `dfs { !it.isColumnGroup() }`. Now it's also present for `rec`, `recursively`.

Reply:
`.rec()` can not only be called on `all()`, you can also call `first { }.rec(includeGroups = false)`, so I think keeping it in rec/recursively too would be best.

Reply:
I'm not sure first + includeGroups will work the way i'd expect. In other cases you can write `colsOf { it.name.startsWith("hi") && !it.isColumnGroup() }.rec()` for the same result. So, parameter doesn't bring anything new. But sometimes it won't make sense: `first { it.type == typeOf<Int> }.rec()`

Reply:
`first { predicate }.rec(includeGroups = false)` returns a `SingleColumn` which will resolve to `null` if there is no non-group column adhering to the predicate, as you'd expect. Recursively modifies what goes into `first`, filtering out groups if needed.
But you're right, it doesn't always add something useful. Let me go over all the uses and see if I can remove it from `recursively` while keeping it for `allRec`

Reply:
Maybe to come up with another name? I think the primary purpose of this function is shorthand to `dfs { !it.isColumnGroup() }` + for newcomers it might not be obvious that (or why) column groups should be included / excluded, so includeGroups = false by default is important, i think

Reply:
I don't think we have a function that can give all valueColumns right? If we did we could have `valueCols {}.recursively()` or `valueColsRecursively()` but that might get a bit long.

Reply:
Plus it's not just ColumnGroups that get included right? also FrameColumns and ValueColumns. So, just all of them. I think that's a good default

Reply:
@koperagen Actually I do want to add `valueCols {}` and `frameCols {}` since we already have `groups {}` as well. Probably in another PR though. Then you can call `valueCols().rec()` if you want. Sounds okay?

Reply:
Could be a nice addition, not 100% related to this problem though. For this function something like "allLeafs" might fit better. I think it was named like that before and renamed later

Reply:
We could also take it in another direction eventually, similar to `colsOf<Type>()` have `colsOfKind(vararg ColumnKind).rec()`, that might be clearer